woocommerce/woocommerce-android
# Issue/9483 add coupon selector vm

Closes: #9483 


### Description

This PR adds the option to select a coupon from the coupon list when the user is creating an order. 

### Testing instructions


1. log into a store that has lots of coupons
2. Go to the orders screen
3. create a new order
4. add a product 
5. once the Add Coupon option is available, click on it
6. make sure the coupon list is displayed
7. click the back arrow and make sure it works
8. click on a coupon and make sure it's added to the order
9. when in the list, make sure the pagination works
10. log into a store that has no coupons
11. make sure the empty coupon list screen is displayed
12. make sure the Go to Coupon list button works and takes you to the Coupon List screen

### Images/gif


| Selection | Empty list |
| ---- | ---- |
| <img src="https://github.com/woocommerce/woocommerce-android/assets/30724184/8685a350-8ca7-403b-af07-1ce99320f067" width="350"/> | <img src="https://github.com/woocommerce/woocommerce-android/assets/30724184/82107ce2-e369-4b34-90aa-0c913c390572" width="350"/> |
